### PR TITLE
Implemented alias action UX.

### PR DESCRIPTION
**Pull request by AWSHurneyt**
### Description
1. Implemented alias action UX.
1. Removed redundant `popout` icons from `Learn more` links. Having `target="_blank"` in the link attributes provides the icon.

### Issues Resolved
https://github.com/opensearch-project/index-management-dashboards-plugin/issues/374

### Screenshots
Blank UX.
Screenshot 2023-05-23 at 11 26 21 AM

UX with valid inputs.
Screenshot 2023-05-23 at 11 04 41 AM

UX with validation errors.
Screenshot 2023-05-23 at 11 21 31 AM

### Check List
- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by AWSHurneyt** [comment]
It looks like the cypress workflow is executing as expected on the 2.x branch. I've created this PR in my fork that executes these checks against the 2.x branch, and all checks succeed.
https://github.com/AWSHurneyt/index-management-dashboards-plugin/pull/2

**Comment by kamingleung** [comment]
@AWSHurneyt 
1. Is there a need to have both add and remove on the same action? Would it be more clear to have separate actions such as "Add aliases" and "Remove aliases"?
2. Can users select from a list of existing aliases to add or remove?

**Comment by AWSHurneyt** [comment]
> @AWSHurneyt
> 
> 1. Is there a need to have both add and remove on the same action? Would it be more clear to have separate actions such as "Add aliases" and "Remove aliases"?
> 2. Can users select from a list of existing aliases to add or remove?

@kamingleung 
1. On the backend, the alias action handles add/remove actions in the same process. In addition, if the user were to interact with the `_alias` API using devtools, they would be able to configure add/remove actions in one command (https://opensearch.org/docs/latest/api-reference/index-apis/alias/); this implementation would preserve that experience in a way. We could separate them into separate UX, but it would be a purely frontend split essentially. I don't think there would be an issue with separating them (aside from the actual implementation time), but @bowenlan-amzn correct me if I'm wrong.
2. At the moment, the combo boxes do not populate with a list of defined aliases. @bowenlan-amzn , would we want to populate the combo boxes with existing aliases as well? I can see the value in that, especially for the remove action.

**Comment by kamingleung** [comment]
@AWSHurneyt 
1. I would recommend having add and remove as separate actions on the UI. This makes it clear and distinctive on what each action is trying to do. It is also clear when reviewing the policy on the UI.

If that cannot be done, I would still consider breaking add / remove in the same flow like this:
image
"Select aliases to add..." and "Select aliases to remove.." fields will be required if users toggled the switches on.

File: public/pages/VisualCreatePolicy/components/UIActions/AliasUIAction.tsx [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+          fullWidth
+          style={{ maxWidth: "100%" }}
+          isInvalid={!this.isValid()}
+          error={this.getAliasActionErrorText(selectedItems).add}
+          data-test-subj={"add-alias-row"}
+        >
+          <>
+            <EuiFormCustomLabel
+              title={"Aliases to add"}
+              helpText={"The provided aliases will be applied to the manage index."}
```
Comment (AWSHurneyt) [inline public/pages/VisualCreatePolicy/components/UIActions/AliasUIAction.tsx:123]: Note to self: this should say "**managed** index."

File: public/pages/VisualCreatePolicy/components/UIActions/AliasUIAction.tsx [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+          fullWidth
+          style={{ maxWidth: "100%" }}
+          isInvalid={!this.isValid()}
+          error={this.getAliasActionErrorText(selectedItems).remove}
+          data-test-subj={"remove-alias-row"}
+        >
+          <>
+            <EuiFormCustomLabel
+              title={"Aliases to remove"}
+              helpText={"The provided aliases will be removed from the manage index."}
```
Comment (AWSHurneyt) [inline public/pages/VisualCreatePolicy/components/UIActions/AliasUIAction.tsx:167]: Note to self: this should say "**managed** index."

**Comment by AWSHurneyt** [comment]
Here are some updated screenshots based on the feedback above.
#### Fields disabled
Screenshot 2023-05-30 at 10 27 50 AM

#### Blank fields
Screenshot 2023-05-30 at 10 28 04 AM

#### Fields with error messages
Screenshot 2023-05-30 at 10 28 56 AM

#### Fields with valid inputs
Screenshot 2023-05-30 at 10 32 12 AM

#### Modal that appears when disabling one of the combo boxes while it contains inputs
Screenshot 2023-05-30 at 10 29 13 AM

**Comment by kamingleung** [comment]
> #### Modal that appears when disabling one of the combo boxes while it contains inputs
> Screenshot 2023-05-30 at 10 29 13 AM

@AWSHurneyt When disabling one of the toggles, instead of asking to clear the field, can we cache the selections, so if the user decides to toggle back on, the selections are still there?

**Comment by AWSHurneyt** [comment]
As mentioned in https://github.com/opensearch-project/index-management-dashboards-plugin/pull/754#issuecomment-1560215249, the cypress test workflow is currently failing in the `main` branch; but in this PR https://github.com/AWSHurneyt/index-management-dashboards-plugin/pull/2, we can see that they succeed when executed against the `2.x` branch.

**Review by amsiglan — APPROVED** [review]

**Review by sbcd90 — APPROVED** [review]

**Review by lezzago — APPROVED** [review]

**Review by bowenlan-amzn — APPROVED** [review]

**Pull request merged**